components/organisms/InsightPage/InsightPage.tsx

```diff
@@ -202,10 +206,9 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
     });
     setCreateLoading(false);
     if (response.ok) {
+      const { id } = await response.json();
       toast({ description: "Page created successfully", variant: "success" });
-      setTimeout(() => {
-        router.push("/hub/insights");
-      }, 1000);
+      router.push(`/pages/${user?.user_metadata.user_name}/${id}/dashboard`);
```

Comment:
Refactor (blocking): If the user is undefined somehow, i.e. `user?`, this will generate an invalid redirect URL.

Consider reviewing the logic to ensure the user does indeed exist before redirecting.